src/main/java/codeu/controller/ConversationServlet.java

```diff
@@ -101,23 +115,46 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
 
     String conversationTitle = request.getParameter("conversationTitle");
     if (!conversationTitle.matches("[\\w*]*")) {
-      request.setAttribute("error", "Please enter only letters and numbers.");
-      request.getRequestDispatcher("/WEB-INF/view/conversations.jsp").forward(request, response);
-      return;
+      if (conversationTitle.contains("#")) {
+        String hashtagTitle = (String)conversationTitle.substring(conversationTitle.indexOf("#") + 1);
```

Comment:
how about moving this line outside of the whole 'if(!conversationTitle.matches...)'?

indexOf will return -1 if the string is not found. If we do substring (-1 + 1 = 0), it will return the original string.

also, why is a string conversion necessary?